### PR TITLE
refactor(astro): code refactor

### PR DESCRIPTION
**Pull request by Simon-He95**
## Changes

refactor: reduce some additional conditional judgments

### CODE REVIEW

**Comment by ematipico** [comment]
@Simon-He95 I will move this PR to draft, until you fix the CI and provide correct description.

**Comment by Simon-He95** [comment]
@ematipico, Can you help me review?

File: packages/astro/src/core/config/merge.ts [hunk 1]
```diff
@@ -36,9 +36,7 @@ function mergeConfigRecursively(
 				};
 				continue;
 			}
-		}
-
-		if (key === 'data' && rootPath === 'db') {
+		} else if (key === 'data' && rootPath === 'db') {
```
Comment (bluwy) [inline packages/astro/src/core/config/merge.ts:39]: Can this be reverted? I think it'd be more consistent with the rest of the code here.

**Comment by ematipico** [comment]
@Simon-He95, please fill out the template and explain your refactors.

**Comment by Simon-He95** [comment]
> @Simon-He95, please fill out the template and explain your refactors.

Refactoring basically reduces extra conditional judgments, because the two conditions are conflicting. I matched A, but it will still be judged once by B's if condition. This refactoring is to solve this problem.

**Comment by ematipico** [comment]
I was mainly asking to **update** the PR description (template) :)

**Comment by Simon-He95** [comment]
> I was mainly asking to **update** the PR description (template) :)

ah, oh :)

**Comment by Simon-He95** [comment]
> I was mainly asking to **update** the PR description (template) :)

ah, ok :)

**Comment by Simon-He95** [comment]
> I was mainly asking to **update** the PR description (template) :)

ah, ok :)

**Review by ematipico — APPROVED** [review]

File: packages/astro/src/core/config/merge.ts (hunk 1 above)
Comment (bluwy) [reply packages/astro/src/core/config/merge.ts:39]: This code block contains top-level if conditions to do certain things, using `else if` means that it's related to the previous `if` statement, which it isn't.

I don't think it's worth changing this and the performance gain isn't a lot.

Comment (Simon-He95) [reply packages/astro/src/core/config/merge.ts:39]: Ah， I get you

**Review by bluwy — APPROVED** [review]

**Pull request merged**